### PR TITLE
fuzz: cap stats utility_fuzz_test iterations

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: Limiting the number of a loop iterations in stats/utility_fuzz_test.
Additional Description: This avoids test timeout when the input is large.
Risk Level: low - test only
Testing: Fixing a fuzz test
Docs Changes: -
Release Notes: -

Signed-off-by: Adi Suissa-Peleg <adip@google.com>

### CODE REVIEW

**Comment by adisuissa** [comment]
Seems like the timeout error in clang_tidy was due to very slow compilation of boringssl.
Can you please restart CI/azp?

File: test/common/stats/utility_fuzz_test.cc [hunk 1]
```diff
@@ -53,7 +61,8 @@ DEFINE_FUZZER(const uint8_t* buf, size_t len) {
     Stats::Utility::counterFromElements(*scope, {});
   } else {
     // add random length string in each loop
-    while (provider.remaining_bytes() > 3) {
+    size_t iteration = 0;
```
Comment (asraa) [inline test/common/stats/utility_fuzz_test.cc:64]: nit: suggest for loop with remaining bytes condition over iteration?

Comment (adisuissa) [reply test/common/stats/utility_fuzz_test.cc]: Done

**Comment by adisuissa** [comment]
@asraa Thanks for reviewing this.

**Pull request merged**